letsencrypt/client/display/util.py

```diff
@@ -0,0 +1,410 @@
+
+        :param str message: title of menu
+        :param choices: Menu lines, len must be > 0
+        :type choices: list of tuples (tag, item) or
+            list of descriptions (tags will be enumerated)
+
+        :returns: tuple of the form (code, tag) where
+            code - int display exit code
+            tag - str corresponding to the item chosen
+        :rtype: tuple
+
+        """
+        self._print_menu(message, choices)
+
+        code, selection = self._get_valid_int_ans(len(choices))
+
+        return code, selection - 1
+
+    def input(self, message):
+        # pylint: disable=no-self-use
+        """Accept input from the user
+
+        :param str message: message to display to the user
+
+        :returns: tuple of (`code`, `input`) where
+            `code` - str display exit code
+            `input` - str of the user's input
+        :rtype: tuple
+
+        """
+        ans = raw_input(
+            textwrap.fill("%s (Enter 'c' to cancel): " % message, 80))
+
+        if ans == "c" or ans == "C":
+            return CANCEL, "-1"
+        else:
+            return OK, ans
+
+    def yesno(self, message, yes_label="Yes", no_label="No"):
+        """Query the user with a yes/no question.
+
+        Yes and No label must begin with different letters
+
+        :param str message: question for the user
+        :param str yes_label: Label of the "Yes" parameter
+        :param str no_label: Label of the "No" parameter
+
+        :returns: True for "Yes", False for "No"
+        :rtype: bool
+
+        """
+        side_frame = ("-" * 79) + os.linesep
+
+        message = self._wrap_lines(message)
+
+        self.outfile.write("{0}{frame}{msg}{0}{frame}".format(
+            os.linesep, frame=side_frame, msg=message))
+
+        while True:
+            ans = raw_input("{yes}/{no}: ".format(
+                yes=_parens_around_char(yes_label),
+                no=_parens_around_char(no_label)))
+
+            # Couldn't get pylint indentation right with elif
+            # elif doesn't matter in this situation
+            if (ans.startswith(yes_label[0].lower()) or
+                    ans.startswith(yes_label[0].upper())):
+                return True
+            if (ans.startswith(no_label[0].lower()) or
+                    ans.startswith(no_label[0].upper())):
+                return False
+
+    def checklist(self, message, tags):
+        """Display a checklist.
+
+        :param str message: Message to display to user
+        :param list tags: `str` tags to select, len(tags) > 0
+
+        :returns: tuple of (`code`, `tags`) where
+            `code` - str display exit code
+            `tags` - list of selected tags
+        :rtype: tuple
+
+        """
+        while True:
+            self._print_menu(message, tags)
+
+            code, ans = self.input("Select the appropriate numbers "
+                                   "separated by commas and/or spaces ")
```

Comment:
you add a space after the message in input(message), so you don't need one at the end here.